src/utils/storage.js

```diff
@@ -145,12 +145,26 @@ export function writeMetadataToDb({
       })
 
       // Function for creating promise for writing file metadata (handles writing to RTDB or Firestore)
+      const documentIdFromOptions =
+        typeof documentId === 'function'
+          ? documentId(
+              uploadTaskSnapshot,
+              firebase,
+              uploadTaskSnapshot.metadata,
+              downloadURL
+            )
+          : documentId
       const metaSetPromise = (fileData) => {
         if (useFirestoreForStorageMeta) {
-          return firebase // Write metadata to Firestore
-            .firestore()
-            .collection(dbPath)
-            .add(fileData)
+          if (documentIdFromOptions) {
+            const docRef = firebase // Write metadata to Firestore
+              .firestore()
+              .collection(dbPath)
+              .doc(documentIdFromOptions)
+            return docRef.update(fileData).then(() => docRef)
```

Comment:
Nice work! Wondering if we should use set with merge instead of update here? Or at least off the option?

Reply:
That's an interesting thought. Using `set` with the merge property functions the same as `update` except that in addition to updating properties on an existing document it will create a new document if the `documentId` does not already exist. I'm thinking `set` with merge should just be the default. This gives the flixibility of being able to create and update documents in bulk and puts the responsibility on the developer to make sure the documentId exists if they don't want a new document created.

I could add another option property but it kind of seems like if you're smart enough to set a flag to fail if your documentId doesn't already exist you're probably smart enough to have filtered it out of your update in the first place 😉.

I'm going to make this change.

Reply:
I just noticed today that the `updateProfile` method has a `useSet` option that is probably what you're talking about here, maybe I will add something like that for consistency's sake.